src/java.base/share/classes/sun/security/ssl/ServerHello.java

```diff
@@ -565,6 +565,27 @@ public byte[] produce(ConnectionContext context,
                     clientHello);
             shc.serverHelloRandom = shm.serverRandom;
 
+            // Key Encapsulation Mechanism (KEM):
```

Comment:
I really like the details you included here, but a little wordsmithing would improve readability.  

Can you please add an intro sentence like:  "For key derivation, we'll either use the KA or a KEM model, depending on what group is used."  

Then add some of the lines from 579-584, then do the KEM and KA sections, and move the final sentence to the KEM section.  

I think this would read much cleaner.

Reply:
Updated the comments. Hope it is more cleaner.